File: python/src/mapper.py
def mapper(block, matrix_id, i, j, n_blocks, m_blocks, p_blocks):
    outputs = []

    if matrix_id == "A":
        for col in range(p_blocks):
            outputs.append(((i, col), ("A", j, block)))
    else:
        for row in range(n_blocks):
            outputs.append(((row, j), ("B", i, block)))

    return outputs

File: java/src/Mapper.java
import java.util.ArrayList;
import java.util.List;

public class Mapper {

    public static List<MapOutput> map(
            double[][] block,
            String matrixId,
            int i,
            int j,
            int nBlocks,
            int pBlocks
    ) {
        List<MapOutput> outputs = new ArrayList<>();

        if (matrixId.equals("A")) {
            for (int col = 0; col < pBlocks; col++) {
                outputs.add(new MapOutput(
                        new BlockIndex(i, col),
                        new BlockValue("A", j, block)
                ));
            }
        } else {
            for (int row = 0; row < nBlocks; row++) {
                outputs.add(new MapOutput(
                        new BlockIndex(row, j),
                        new BlockValue("B", i, block)
                ));
            }
        }
        return outputs;
    }
}
